GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp

```diff
@@ -772,6 +767,11 @@ void Renderer::renderMaterialShader(RenderState& rstate, RenderData* render_data
                 error.c_str());
         shader_manager->getErrorShader()->render(&rstate, render_data, curr_material);
     }
+    programId = shader->getProgramId();
+    if (programId != -1)
```

Comment:
It is a sanity check. This case should not occur. I can leave this as is or trigger an error in generateVAO() for invalid programId.